R/readers.R

```diff
@@ -309,4 +309,29 @@ read_s3 <- function(file){
   data <- readr::read_log(file = file, col_types = types, col_names = names)
   data$request_time <- readr::parse_datetime(data$request_time, format = "%d/%b/%Y:%H:%M:%S %z")
   return(data)
-}
\ No newline at end of file
+}
+
+#'@title Read Microsoft IIS Logs
+#'@description \code{read_iis} provides a reader for Microsoft IIS access logs.
+#'
+#'@param file the full path to the IIS log file you want to read.
+#'
+#'@examples
+#'# Using the inbuilt testing dataset
+#'iis_data <- read_iis(system.file(file.path("extdata", "iis.log"), package = "webreadr"))
+#'
+#'@export
+
+read_iis <- function(file){
+  #read in the text file to search for server restarts
+  raw_file <- readLines(file)
+  mask <- !substr(raw_file, 1, 1)=="#"  
```

Comment:
Are there a specified number of commented lines each time? If so, a skip command might work more efficiently for large files.

Reply:
I'm working with lots of logs and this was one of those unfortunate 'in the wild' issues that I ran into.  It's not super common, but it's very frustrating - basically, if an IIS server ever re-starts in the middle of the day, it opens up the existing log file and writes the header before it writes any data - it's at the top of every log, but it also can be in the _middle_ of any log if there was a server restart.

Reply:
....good _gravy_. Okay, that makes total sense, although I might play around with a couple of ways of getting at that to test speed.